src/Socket.php

```diff
@@ -255,13 +255,14 @@ public function read($length, $type = PHP_BINARY_READ)
      *
      * @param int $length maximum length to read
      * @param int $flags
+     * @param int $ref reference will be filled with the length of data read from the socket
      * @return string
      * @throws Exception on error
      * @see self::read() if you do not need to pass $flags
      * @see self::recvFrom() if your socket is not connect()ed
      * @uses socket_recv()
      */
-    public function recv($length, $flags)
+    public function recv($length, $flags, &$ret)
```

Comment:
I am not sure if you want me to set this to null by default (which would make it not break current implementations of it).